src/Modules/Trinity.DynamicCluster/Trinity.DynamicCluster/DynamicMemoryCloud/DynamicMemoryCloud.Cmd.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Graph Engine
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+using Trinity;
+using Trinity.Core.Lib;
+using Trinity.Network.Messaging;
+using Trinity.Network;
+
+namespace Trinity.DynamicCluster.Storage
+{
+    public partial class DynamicMemoryCloud
+    {
+        /// <summary>
+        /// Dumps memory storages to disk files on all Trinity servers.
+        /// </summary>
+        /// <returns>true if saving succeeds; otherwise, false.</returns>
+        public override bool SaveStorage()
+        {
+            throw new NotImplementedException();
```

Comment:
new save format, persistency adapters, and a storage iterator that does not lock up the whole local storage...